### PR TITLE
S2U-14 5.1.1.10 T&Q: Canceled questions and score re-calculation - MASTER

### PR DESCRIPTION
**Pull request by V100X**
Jira [S2U-14](https://sakaiproject.atlassian.net/browse/S2U-14)
Developed by: @stetsche

This is a port from Sakai 22 to 24

From PRs: (#11328) (#11542)

[S2U-14]: https://sakaiproject.atlassian.net/browse/S2U-14?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by mpellicer** [comment]
Please could you add a reference PR for the conversion scripts ?

**Review by mpellicer — CHANGES_REQUESTED** [review]
conversion scripts needed.

**Review by mpellicer — APPROVED** [review]
Thanks for the hard work @V100X and @stetsche

**Review by mpellicer — CHANGES_REQUESTED** [review]
Seems that @stetsche has spotted a bug present on this, can you explain it Markus? Is it affecting 22.x ?

**Comment by mpellicer** [comment]
Thank you very much for reporting it!

**Comment by V100X** [comment]
Should be ready now 👍

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/RepublishAssessmentListener.java [hunk 1]
```diff
@@ -216,126 +216,4 @@ private void postUserNotification(PublishedAssessmentFacade assessment, Publishe
...
-          GradebookFacade.getGradebookUId(),
-          assessment.getPublishedAssessmentId().toString(), g);
-      }
-      catch(Exception e){
-        log.info("*** oh well, looks like there is nothing to remove:"+e.getMessage());
-      }
-    }
-	}
-}
+}
```
Comment (stetsche) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/RepublishAssessmentListener.java:219]:
```suggestion
}

```

Comment (stetsche) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/RepublishAssessmentListener.java:219]:
```suggestion
}
```

**Review by stetsche — APPROVED** [review]
Yeah, looks all good now, well done @V100X!

**Review by mpellicer — APPROVED** [review]
Awesome! thanks for the effort guys

**Pull request merged**